base/iterators.jl

```diff
@@ -1595,4 +1595,72 @@ end
 # be the same as the keys, so this is a valid optimization (see #51631)
 pairs(s::AbstractString) = IterableStatePairs(s)
 
+"""
+    nth(itr, n::Integer)
+
+Get the `n`th element of an iterable collection. Return `nothing` if not existing.
+
```

Comment:
I would also mention that it will advance the iterator `n` steps, and perhaps show an example of how it affects a stateful iterator in the examples:
```
julia> stateful = Stateful(1:10); nth(stateful, 7)
7

julia> first(stateful)
8
```